kdbg/dbgmainwnd.cpp

```diff
@@ -514,6 +514,9 @@ void DebuggerMainWnd::restoreSettings(KSharedConfigPtr config)
 
     m_isX86 = isTargetOfFileX86(m_target);
     m_asmGlobalFlavor = pg.readEntry(DisassemblyFlavor, m_asmGlobalFlavor);
+    //If the setting entry is empty, set an initial value
+    if(m_asmGlobalFlavor.isEmpty())
+	m_asmGlobalFlavor = DefaultGenericFlavor;
```

Comment:
This would not be necessary if the default flavor were just the empty string. That the user sees "default" in the dialog would than just be a matter of presentation that the dialog(s) would have to take care of.

Other changes in this patch are either unnecessary churn or of the kind "oops, let's fix an earlier mistake", which should be squashed into the earlier commit as if the mistake had never happened.